file_watcher/lastrun_file_monitor.py

```diff
@@ -44,38 +52,107 @@ def __init__(
-            and int(self.latest_known_run_from_db) < int(self.last_recorded_run_from_file)
+            and int(self.latest_known_run_from_fia) < int(self.last_recorded_run_from_file)
         ):
             logger.info(
                 "Recovering lost runs between %s and%s",
                 self.last_recorded_run_from_file,
-                self.latest_known_run_from_db,
+                self.latest_known_run_from_fia,
+            )
+            self.recover_lost_runs(self.latest_known_run_from_fia, self.last_recorded_run_from_file)
+            self.latest_known_run_from_fia = self.last_recorded_run_from_file
+
+    def retry_api_request(
+        self,
+        url_request_string: str,
+        method: Literal["GET", "POST", "PATCH", "PUT", "DELETE"],
+        body: dict[Any, Any] | None = None,
+        retry_attempts: int = 5,
+    ) -> requests.Response:
+        """
+        A helper function to handle multiple attempts at HTTP Requests
+        :param url_request_string: url of FIA API to get/put
+        :param method: GET or POST
+        :param body: dictionary to POST
+        :param retry_attempts: number of times to retry the request
+        :return: a requests.Response object for handling
+        """
+        attempts = 0
+        auth = HTTPBasicAuth("apikey", self.fia_api_api_key)
+        while attempts < retry_attempts:
+            req = requests.request(
+                method=method, url=url_request_string, timeout=self.request_timeout_length, auth=auth, json=body
             )
-            self.recover_lost_runs(self.latest_known_run_from_db, self.last_recorded_run_from_file)
-            self.latest_known_run_from_db = self.last_recorded_run_from_file
+            attempts += 1
+            if req.status_code == HTTPStatus.OK:
+                return req
+            if retry_attempts == attempts:
+                return req
+            # increment sleep time as retries persist
+            time.sleep(5 * self.request_timeout_length)
+
+        # if retries failed, try one last time and return a response object for handling
+        return requests.request(method=method, url=url_request_string, timeout=30)
 
-    def get_latest_run_from_db(self) -> Union[str, None]:
+    def get_latest_run_from_fia_api(self, instrument: str) -> str | None:
         """
-        Retrieve the latest run from the database
-        :return: Return the latest run for the instrument that is set on this object
+        Retrieve the latest run using FIA API
+        :param instrument: name of instrument
+        :return: Return the latest run for the instrument that is set on this objector or None if unsuccessful
         """
-        # This likely contains NDX<INSTNAME> so remove the NDX and go for it with the DB
-        actual_instrument = self.instrument[3:]
-        return self.db_updater.get_latest_run(actual_instrument)
+        # Use request library to FIA API
+        instrument_name = instrument
```

Comment:
In the previous code, it stripped the first 3 chars - in case it started with NDX. Is this still necessary? Would it be worth adding a check to make sure something like:
```suggestion
        instrument_name = instrument if not instrument.startswith("NDX".lower()) else instrument[3:]
```